ILRepack/RepackImporter.cs

```diff
@@ -156,31 +160,41 @@ public TypeDefinition Import(TypeDefinition type, Collection<TypeDefinition> col
             _repackContext.MappingHandler.StoreRemappedType(type, nt);
 
             // nested types first (are never internalized)
-            foreach (TypeDefinition nested in type.NestedTypes)
+            foreach (TypeDefinition nested in type.NestedTypes.Where(CanInclude))
                 Import(nested, nt.NestedTypes, false);
-            foreach (FieldDefinition field in type.Fields)
+            foreach (FieldDefinition field in type.Fields.Where(CanInclude))
                 CloneTo(field, nt);
 
             // methods before fields / events
-            foreach (MethodDefinition meth in type.Methods)
+            foreach (MethodDefinition meth in type.Methods.Where(CanInclude))
                 CloneTo(meth, nt, justCreatedType);
 
-            foreach (EventDefinition evt in type.Events)
+            foreach (EventDefinition evt in type.Events.Where(CanInclude))
                 CloneTo(evt, nt, nt.Events);
-            foreach (PropertyDefinition prop in type.Properties)
+            foreach (PropertyDefinition prop in type.Properties.Where(CanInclude))
                 CloneTo(prop, nt, nt.Properties);
             return nt;
         }
 
-        // Real stuff below //
-        // These methods are somehow a merge between the clone methods of Cecil 0.6 and the import ones of 0.9
-        // They use Cecil's MetaDataImporter to rebase imported stuff into the new assembly, but then another pass is required
-        //  to clean the TypeRefs Cecil keeps around (although the generated IL would be kind-o valid without, whatever 'valid' means)
-
-        /// <summary>
-        /// Clones a field to a newly created type
-        /// </summary>
-        private void CloneTo(FieldDefinition field, TypeDefinition nt)
+		private bool CanInclude<TMember>(TMember member) where TMember : ICustomAttributeProvider, IMemberDefinition {
```

Comment:
Could be renamed to `ShouldDropForRepacking`?

Reply:
'Cause it is used in the `Where` filter. If we use a negative method, then we have to rewrite the `Where` clauses with a lambda expression.

Reply:
OK, I renamed that method and use an `if` in each place to filter the result where `CloneTo` was called